y/y.go

```diff
@@ -179,6 +183,30 @@ func FixedDuration(d time.Duration) string {
 	return str
 }
 
+// BuildChecksum builds sha256 checksum for the given data.
+func BuildChecksum(data []byte) ([]byte, error) {
+	sum := sha256.New()
+	bytesWritten, err := sum.Write(data)
+	if bytesWritten != len(data) {
+		return nil, errors.New("Size mismatch between bytes written and length of data")
+	}
+	if err != nil {
```

Comment:
Should check the error first before checking the `bytesWritten`

Reply:
Nice catch. Thanks. 
Fixed in https://github.com/dgraph-io/badger/pull/816/commits/fc950be2fe51ca589fb19c77e33bf10cafb9c1bd